src/modules/microdds_client/microdds_client.cpp

```diff
@@ -142,58 +178,73 @@ void MicroddsClient::run()
 				       sizeof(output_data_stream_buffer));
 
 		uint8_t input_reliable_stream_buffer[BUFFER_SIZE] {};
-		uxr_create_input_reliable_stream(&session, input_reliable_stream_buffer, BUFFER_SIZE, STREAM_HISTORY);
+		uxrStreamId reliable_in = uxr_create_input_reliable_stream(&session, input_reliable_stream_buffer,
+					  sizeof(input_reliable_stream_buffer),
+					  STREAM_HISTORY);
+
 		uxrStreamId input_stream = uxr_create_input_best_effort_stream(&session);
 
 		// Create entities
 		uxrObjectId participant_id = uxr_object_id(0x01, UXR_PARTICIPANT_ID);
-		const char *participant_xml = _localhost_only ?
-					      "<dds>"
-					      "<profiles>"
-					      "<transport_descriptors>"
-					      "<transport_descriptor>"
-					      "<transport_id>udp_localhost</transport_id>"
-					      "<type>UDPv4</type>"
-					      "<interfaceWhiteList><address>127.0.0.1</address></interfaceWhiteList>"
-					      "</transport_descriptor>"
-					      "</transport_descriptors>"
-					      "</profiles>"
-					      "<participant>"
-					      "<rtps>"
-					      "<name>default_xrce_participant</name>"
-					      "<useBuiltinTransports>false</useBuiltinTransports>"
-					      "<userTransports><transport_id>udp_localhost</transport_id></userTransports>"
-					      "</rtps>"
-					      "</participant>"
-					      "</dds>"
-					      :
-					      "<dds>"
-					      "<participant>"
-					      "<rtps>"
-					      "<name>default_xrce_participant</name>"
-					      "</rtps>"
-					      "</participant>"
-					      "</dds>" ;
-		uint16_t participant_req = uxr_buffer_create_participant_xml(&session, reliable_out, participant_id, 0,
-					   participant_xml, UXR_REPLACE);
+
+		uint16_t domain_id = 0;
+
+		// const char *participant_xml = _localhost_only ?
+		// 			      "<dds>"
+		// 			      "<profiles>"
+		// 			      "<transport_descriptors>"
+		// 			      "<transport_descriptor>"
+		// 			      "<transport_id>udp_localhost</transport_id>"
+		// 			      "<type>UDPv4</type>"
+		// 			      "<interfaceWhiteList><address>127.0.0.1</address></interfaceWhiteList>"
+		// 			      "</transport_descriptor>"
+		// 			      "</transport_descriptors>"
+		// 			      "</profiles>"
+		// 			      "<participant>"
+		// 			      "<rtps>"
+		// 			      "<name>default_xrce_participant</name>"
+		// 			      "<useBuiltinTransports>false</useBuiltinTransports>"
+		// 			      "<userTransports><transport_id>udp_localhost</transport_id></userTransports>"
+		// 			      "</rtps>"
+		// 			      "</participant>"
+		// 			      "</dds>"
+		// 			      :
+		// 			      "<dds>"
+		// 			      "<participant>"
+		// 			      "<rtps>"
+		// 			      "<name>default_xrce_participant</name>"
+		// 			      "</rtps>"
+		// 			      "</participant>"
+		// 			      "</dds>" ;
+		// uint16_t participant_req = uxr_buffer_create_participant_xml(&session, reliable_out, participant_id, domain_id, participant_xml, UXR_REPLACE);
+
+		const char *participant_name = "px4_xrce_participant";
```

Comment:
I would include the participant namespace here. It is not mandatory for this to work but when you use [micro-ROS-Agent](https://github.com/micro-ROS/micro-ROS-Agent) it creates two nodes with the same name in ROS2 and a lot of tools in ROS2 don't like this.

Somethin uneder the form namespace/px4_xrce_participant.